tests/Unit/MutationTest.php

```diff
@@ -123,6 +92,36 @@ public function testValidationError(): void
 
         $attributes = $field->getAttributes();
 
+        try {
+            $attributes['resolve'](null, [
+                'test_with_rules_non_nullable_input_object' => [
+                    'val' => 4,
+                ],
+            ], [], null);
+        } catch (ValidationError $e) {
+            $validator = $e->getValidator();
+
+            $this->assertInstanceOf(Validator::class, $validator);
+
+            $messages = $e->getValidatorMessages();
+
+            $this->assertTrue($messages->has('test'));
+            $this->assertTrue($messages->has('test_with_rules'));
+            $this->assertTrue($messages->has('test_with_rules_closure'));
+            $this->assertTrue($messages->has('test_with_rules_non_nullable_input_object.otherValue'));
+            $this->assertTrue($messages->has('test_with_rules_non_nullable_input_object.nest'));
+            $this->assertTrue($messages->has('test_with_rules_non_nullable_input_object.list'));
+            $this->assertCount(6, $messages->all());
+        }
+    }
+
+    public function testWithEmptyInput(): void
+    {
+        $class = $this->getFieldClass();
+        $field = new $class();
+
+        $attributes = $field->getAttributes();
+
         try {
             $attributes['resolve'](null, [], [], null);
         } catch (ValidationError $e) {
```

Comment:
I see. The feedback from the last test: this pattern was already here.

Please fix it here, too.